### PR TITLE
Vue training - 02 - Vue components

### PR DESCRIPTION
**Pull request by rsuescun**
## Summary

-`TechTitle` and `TechBox` vue components was added

## Screenshots
Screen Shot 2019-04-23 at 2 52 57 PM

## Trello Card

https://trello.com/c/SsAkTURB/15-vue-components

### CODE REVIEW

File: vue-training/src/components/TechBox.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<template lang="pug">
+  div.tech-box
```
Comment (FrankIglesias) [inline vue-training/src/components/TechBox.vue]: remove the div tag just leave `.tech-box`

File: vue-training/src/components/TechBox.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    p.tech-box-text
+      | {{ tech.description }}
+</template>
+<script>
+export default {
+  props: {
+    tech: {
+      type: Object,
+      required: true,
+      validator: value => value && value.title && value.launch && value.description && value.logo
```
Comment (FrankIglesias) [inline vue-training/src/components/TechBox.vue:19]: It's ok this, we are not used to doing this. When your component grows you will have an enormous function

File: vue-training/src/components/TechTitle.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    align-items: center;
+    color: $black;
+    display: flex;
+    font-size: $font-title-size;
+    margin: 80px;
+    padding: 0 50px;
+    position: relative;
+
+    &::before{
+      content: url('http://localhost:3000/assets/triangle.svg');
```
Comment (FrankIglesias) [inline vue-training/src/components/TechTitle.vue]: delete localhost here

File: vue-training/src/components/TechTitle.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  .title {
+    align-items: center;
+    color: $black;
+    display: flex;
+    font-size: $font-title-size;
+    margin: 80px;
+    padding: 0 50px;
+    position: relative;
+
+    &::before{
```
Comment (FrankIglesias) [inline vue-training/src/components/TechTitle.vue]: space before curly braces

File: vue-training/src/components/TechTitle.vue [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      content: url('http://localhost:3000/assets/triangle.svg');
+      height: $title-decoration-height;
+      left: 0px;
```
Comment (FrankIglesias) [inline vue-training/src/components/TechTitle.vue]: remove px from 0

File: vue-training/src/scss/base.scss [hunk 6]
```diff
@@ -14,9 +14,30 @@
...
   flex-direction: column;
-  grid-row: 2;
   padding: 20px;
 }
+
+.tech-box-list {
+  align-items: start;
+  align-self: center;
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
```
Comment (FrankIglesias) [inline vue-training/src/scss/base.scss]: repeat here

File: vue-training/src/scss/base.scss [hunk 7]
```diff
@@ -14,9 +14,30 @@
...
+  grid-template-rows: 1fr 1fr;
```
Comment (FrankIglesias) [inline vue-training/src/scss/base.scss]: same

File: vue-training/src/views/index/index.pug [hunk 8]
```diff
@@ -11,69 +11,28 @@ html(lang='en')
...
-          span.tech-box-launch Release: 2016
-          p.tech-box-text
-            | ASP.NET Core is the next generation of ASP.NET. It is a modular framework that runs on both the full .NET Framework, on Windows, and the cross-platform .NET Core.
-        .tech-box
-          img.tech-box-logo(src='assets/tech/django_icon.svg' alt='Django logo')
-          h3.tech-box-title Django
-          span.tech-box-launch Release: 2005
-          p.tech-box-text
-            | Django (/ˈdʒæŋɡoʊ/ JANG-goh)[5] is a free and open-source web framework, written in Python, which follows the model-view-template (MVT) architectural pattern.
+      div.tech-list(
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: remove div here

File: vue-training/src/views/index/index.pug [hunk 9]
```diff
@@ -11,69 +11,28 @@ html(lang='en')
     script(src='./vendor.js')
   body.app-container
     nav.navbar
-      img.navbar-logo(src='assets/wolox_logo.svg' alt='App logo')
+      img.navbar-logo(
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: move all this inside the #app.main div

File: vue-training/src/views/index/index.pug [hunk 10]
```diff
@@ -11,69 +11,28 @@ html(lang='en')
...
-          h3.tech-box-title Django
-          span.tech-box-launch Release: 2005
-          p.tech-box-text
-            | Django (/ˈdʒæŋɡoʊ/ JANG-goh)[5] is a free and open-source web framework, written in Python, which follows the model-view-template (MVT) architectural pattern.
+      div.tech-list(
+        v-for="techGroup in techs"
+        :key="techGroup.id"
+      )
+        tech-title(:title="techGroup.title")
+        div.tech-box-list
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.pug]: remove this

File: vue-training/src/views/index/index.js [hunk 11]
```diff
@@ -2,12 +2,98 @@ import Vue from 'vue'
...
-  el: '#app'
+  el: '#app',
+  components: {
+    TechBox,
+    TechTitle
+  },
+  data: {
+    techs: [
+      {
+        id: 1,
```
Comment (FrankIglesias) [inline vue-training/src/views/index/index.js]: you can delete this and use the index inside the v-for (think that here frontend-Tools and reactjs have the same id 😉

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: vue-training/src/components/TechBox.vue (hunk 2 above)
Comment (rsuescun) [reply vue-training/src/components/TechBox.vue:19]: How can you easily validate that the object complies with a scheme?

Comment (FrankIglesias) [reply vue-training/src/components/TechBox.vue:19]: Using typescript for instance

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**